app/app.js

```diff
@@ -475,98 +486,87 @@ angular.module('app', ['angular-md5'])
+    *
+    * @see https://github.com/kayellpeee/hsl_rgb_converter
     */
-    var chroma = (1 - Math.abs((2 * lightness) - 1)) * saturation;
-    var huePrime = hue / 60;
-    var secondComponent = chroma * (1 - Math.abs((huePrime % 2) - 1));
+    chroma = (1 - Math.abs((2 * lightness) - 1)) * saturation;
+    huePrime = hue / 60;
+    secondComponent = chroma * (1 - Math.abs((huePrime % 2) - 1));
 
     huePrime = Math.floor(huePrime);
-    var red;
-    var green;
-    var blue;
-
-    if( huePrime === 0 ){
-      red = chroma;
-      green = secondComponent;
-      blue = 0;
-    }else if( huePrime === 1 ){
-      red = secondComponent;
-      green = chroma;
-      blue = 0;
-    }else if( huePrime === 2 ){
-      red = 0;
-      green = chroma;
-      blue = secondComponent;
-    }else if( huePrime === 3 ){
-      red = 0;
-      green = secondComponent;
-      blue = chroma;
-    }else if( huePrime === 4 ){
-      red = secondComponent;
-      green = 0;
-      blue = chroma;
-    }else if( huePrime === 5 ){
-      red = chroma;
-      green = 0;
-      blue = secondComponent;
+
+    switch (huePrime){
+      case 0:
+        red = chroma;
+        green = secondComponent;
+        break;
+      case 1:
+        red = secondComponent;
+        green = chroma;
+        break;
+      case 2:
+        green = chroma;
+        blue = secondComponent;
+        break;
+      case 3:
+        green = secondComponent;
+        blue = chroma;
+        break;
+      case 4:
+        red = secondComponent;
+        blue = chroma;
+        break;
+      case 5:
+        red = chroma;
+        blue = secondComponent;
     }
 
-    var lightnessAdjustment = lightness - (chroma / 2);
+
+    lightnessAdjustment = lightness - (chroma / 2);
     red += lightnessAdjustment;
     green += lightnessAdjustment;
     blue += lightnessAdjustment;
 
-    var rgbColor = [Math.round(red * 255), Math.round(green * 255), Math.round(blue * 255)];
+    rgbColor = [Math.round(red), Math.round(green), Math.round(blue)];
 
-    /*
-     Given a color in RGB format, assign either a light
-     or dark color.
-
-     @see https://stackoverflow.com/questions/3942878/how-to-decide-font-color-in-white-or-black-depending-on-background-color
+    /**
+     * Given a color in RGB format, assign either a light
+     * or dark color.
+     *
+     * @see https://stackoverflow.com/questions/3942878/how-to-decide-font-color-in-white-or-black-depending-on-background-color
      */
-      var C, L;
 
-      var R = rgbColor[0];
-      var G = rgbColor[1];
-      var B = rgbColor[2];
-
-      C = [ R/255, G/255, B/255 ];
-
-      for (var i = 0; i < C.length; ++i) {
-        if (C[i] <= 0.03928) {
-          C[i] = C[i] / 12.92;
-        } else {
-          C[i] = Math.pow( ( C[i] + 0.055 ) / 1.055, 2.4);
-        }
-      }
-
-      L = 0.2126 * C[0] + 0.7152 * C[1] + 0.0722 * C[2];
-
-      if (L > 0.179) {
-        textColor = '#444';
+    for (var i = 0; i < rgbColor.length; ++i) {
```

Comment:
We can change `rgbColor.length` to `3` in this case to make it easy. Also in general `for` loops shouldn't have `.length` inside them since it is computed each time, so even for arrays with dynamic lengths it's best to assign it to a variable outside and reference that.